src/Components/Endpoints/src/PublicAPI.Unshipped.txt

```diff
@@ -33,23 +62,40 @@ Microsoft.AspNetCore.Components.Endpoints.RazorComponentResult<TComponent>.Razor
+Microsoft.AspNetCore.Components.PageComponentBuilder.Equals(Microsoft.AspNetCore.Components.PageComponentBuilder? other) -> bool
+Microsoft.AspNetCore.Components.PageComponentBuilder.HasSource(string! source) -> bool
+Microsoft.AspNetCore.Components.PageComponentBuilder.PageComponentBuilder() -> void
+Microsoft.AspNetCore.Components.PageComponentBuilder.PageType.get -> System.Type!
+Microsoft.AspNetCore.Components.PageComponentBuilder.PageType.set -> void
+Microsoft.AspNetCore.Components.PageComponentBuilder.RouteTemplates.get -> System.Collections.Generic.IReadOnlyList<string!>!
+Microsoft.AspNetCore.Components.PageComponentBuilder.RouteTemplates.set -> void
+Microsoft.AspNetCore.Components.PageComponentBuilder.Source.get -> string!
+Microsoft.AspNetCore.Components.PageComponentBuilder.Source.set -> void
+Microsoft.AspNetCore.Components.PageComponentInfo
+Microsoft.AspNetCore.Components.PageComponentInfo.DisplayName.get -> string!
+Microsoft.AspNetCore.Components.PageComponentInfo.Metadata.get -> System.Collections.Generic.IReadOnlyList<object!>!
+Microsoft.AspNetCore.Components.PageComponentInfo.Route.get -> string!
+Microsoft.AspNetCore.Components.PageComponentInfo.Type.get -> System.Type!
 Microsoft.AspNetCore.Components.PersistedStateSerializationMode
 Microsoft.AspNetCore.Components.PersistedStateSerializationMode.Infer = 1 -> Microsoft.AspNetCore.Components.PersistedStateSerializationMode
 Microsoft.AspNetCore.Components.PersistedStateSerializationMode.Server = 2 -> Microsoft.AspNetCore.Components.PersistedStateSerializationMode
 Microsoft.AspNetCore.Components.PersistedStateSerializationMode.WebAssembly = 3 -> Microsoft.AspNetCore.Components.PersistedStateSerializationMode
 Microsoft.AspNetCore.Components.RazorComponentApplication
-Microsoft.AspNetCore.Components.RazorComponentApplication.Pages.get -> System.Collections.Generic.IEnumerable<Microsoft.AspNetCore.Components.PageDefinition!>!
+Microsoft.AspNetCore.Components.RazorComponentApplication.Components.get -> System.Collections.Generic.IReadOnlyList<Microsoft.AspNetCore.Components.ComponentInfo!>!
+Microsoft.AspNetCore.Components.RazorComponentApplication.Pages.get -> System.Collections.Generic.IReadOnlyList<Microsoft.AspNetCore.Components.PageComponentInfo!>!
 Microsoft.Extensions.DependencyInjection.RazorComponentsServiceCollectionExtensions
+override Microsoft.AspNetCore.Components.ComponentBuilder.Equals(object? obj) -> bool
+override Microsoft.AspNetCore.Components.ComponentBuilder.GetHashCode() -> int
+override Microsoft.AspNetCore.Components.PageComponentBuilder.Equals(object? obj) -> bool
+override Microsoft.AspNetCore.Components.PageComponentBuilder.GetHashCode() -> int
 static Microsoft.AspNetCore.Builder.RazorComponentsEndpointRouteBuilderExtensions.MapRazorComponents<TRootComponent>(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder! endpoints) -> Microsoft.AspNetCore.Builder.RazorComponentEndpointConventionBuilder!
```

Comment:
It's a lot of public API :)

Am I right in thinking the reason it's public is because sourcegenerated code has to interact with it? Do we expect application authors to call these APIs from their own code?

If we're only interested in enabling source generation here, perhaps we could consider some scheme for keeping as much of it as possible in some `CompilerServices` namespace and stating that it is not intended to be used from application code, like we do for some of the Razor Compiler-focused APIs.

Reply:
I've reduced the API here to what is needed by the source generator.